src/adapter/etl-adapter-excel/tests/Flow/ETL/Adapter/Excel/Tests/Unit/ExcelExtractorTest.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\Excel\Tests\Unit;
+
+use function Flow\ETL\Adapter\Excel\DSL\from_excel;
+use Flow\ETL\Exception\InvalidArgumentException;
+use Flow\ETL\Tests\FlowTestCase;
+
+final class ExcelExtractorTest extends FlowTestCase
+{
+    public function test_invalid_sheet_name() : void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Sheet name must be a valid Excel sheet name');
+
+        from_excel(__DIR__ . '/../Fixtures/unknown')
+            ->withOnlySheet('This is veeeeeery long excel sheet name, longer than 32 characters');
+    }
```

Comment:
I guess there could be also a test to validate the exception thrown when path is not local